src/SprykerEco/Zed/AvalaraTax/Business/Calculator/AbstractCartItemAvalaraTaxCalculator.php

```diff
@@ -0,0 +1,154 @@
+
+        $this->executeCreateTransactionRequestAfterPlugins($calculableObjectTransfer, $avalaraCreateTransactionResponseTransfer);
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\CalculableObjectTransfer $calculableObjectTransfer
+     *
+     * @return bool
+     */
+    abstract protected function hasShipmentAddress(CalculableObjectTransfer $calculableObjectTransfer): bool;
+
+    /**
+     * @param \ArrayObject|\Generated\Shared\Transfer\ItemTransfer[] $itemTransfers
+     * @param \Generated\Shared\Transfer\AvalaraCreateTransactionResponseTransfer $avalaraCreateTransactionResponseTransfer
+     *
+     * @return void
+     */
+    abstract protected function calculateTaxForItemTransfers(
+        ArrayObject $itemTransfers,
+        AvalaraCreateTransactionResponseTransfer $avalaraCreateTransactionResponseTransfer
+    ): void;
+
+    /**
+     * @param \Generated\Shared\Transfer\CalculableObjectTransfer $calculableObjectTransfer
+     * @param \Generated\Shared\Transfer\AvalaraCreateTransactionResponseTransfer $avalaraCreateTransactionResponseTransfer
+     *
+     * @return void
+     */
+    protected function executeCreateTransactionRequestAfterPlugins(
+        CalculableObjectTransfer $calculableObjectTransfer,
+        AvalaraCreateTransactionResponseTransfer $avalaraCreateTransactionResponseTransfer
+    ): void {
+        foreach ($this->createTransactionRequestAfterPlugins as $createTransactionRequestAfterPlugin) {
+            $calculableObjectTransfer = $createTransactionRequestAfterPlugin->execute($calculableObjectTransfer, $avalaraCreateTransactionResponseTransfer);
+        }
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\ItemTransfer $itemTransfer
+     * @param \Generated\Shared\Transfer\AvalaraTransactionLineTransfer $avalaraTransactionLineTransfer
+     *
+     * @return void
+     */
+    protected function calculateItemTaxByAvalaraTransactionLineItem(
+        ItemTransfer $itemTransfer,
+        AvalaraTransactionLineTransfer $avalaraTransactionLineTransfer
+    ): void {
+        $taxRate = $this->calculateAvalaraTransactionLineTaxRate($avalaraTransactionLineTransfer->getDetailsOrFail());
+        $taxAmount = $this->moneyFacade->convertDecimalToInteger($avalaraTransactionLineTransfer->getTaxOrFail()->toFloat());
+
+        $itemTransfer
+            ->setTaxRate($taxRate)
+            ->setSumTaxAmount($taxAmount);
+    }
+
+    /**
+     * @param string $transactionLineDetails
+     *
+     * @return float
+     */
+    protected function calculateAvalaraTransactionLineTaxRate(string $transactionLineDetails): float
+    {
+        $taxRateSum = 0.0;
+
+        /** @var \Avalara\TransactionLineDetailModel[] $transactionLineDetailsDecoded */
+        $transactionLineDetailsDecoded = $this->utilEncodingService->decodeJson($transactionLineDetails, false);
+        foreach ($transactionLineDetailsDecoded as $transactionLineDetail) {
+            $taxRateSum += $transactionLineDetail->rate ?? 0.0;
```

Comment:
* Tax rate has to be multiplied by `100` in order to convert it to the percents that Spryker uses.